katdal/datasources.py

```diff
@@ -359,13 +396,11 @@ def _upgrade_flags(chunk_info, telstate):
         logger.debug('No additional flag capture streams found: %s', e)
         return chunk_info
     for s in archived_streams:
-        telstate_s = telstate.root().view(s)
-        if telstate_s['stream_type'] != 'sdp.flags' or \
-           stream_name not in telstate_s['src_streams']:
+        telstate_cs = view_capture_stream(telstate, capture_block_id, s)
+        if telstate_cs['stream_type'] != 'sdp.flags' or \
```

Comment:
I guess this assumes that stream `s` has an explicit `stream_type` key or only inherits from streams of the same type (is that how it works?).

Reply:
Basically yes. I would think inheriting from a different type would be a bad idea though.